src/modules/select/hooks/use-input.js

```diff
@@ -1,17 +1,17 @@
 import { useEffect, useRef, useState } from 'react'
 
-const useInput = (initialValue, { onInput, onChange }) => {
+const useInput = (value, { onInput, onChange }) => {
   const inputRef = useRef(null)
-  const [inputData, changeInputData] = useState({ value: initialValue })
+  const [inputData, changeInputData] = useState({ value })
   const [isInputFocused, setIsInputFocused] = useState(null)
 
   useEffect(() => {
-    changeInputData({ value: inputData.value })
-  }, [initialValue])
+    if (value !== inputData.value) changeInputData({ value })
+  }, [value])
```

Comment:
Why don't you blindly pass it down?

Reply:
To prevent calling that callback several times with the same value